Practice/pig_latin.py

```diff
@@ -2,81 +2,75 @@
 FIRST_LETTER_VOWELS = 'AEIOUaeiou'
 OTHER_LETTER_VOWELS = 'aeiouy'
 PUNCTUATION = ' ,_-.?!:;'
-print('Welcome to the English to pig latin translator')
-print('Please enter a sentence: ')
-sentence = input()
+def convert_to_list(sentence):
+    """Takes a sentence and converts into a list"""
+    word_list = sentence.split()
+    return word_list
 
-word_list = sentence.split()
 
-letter_list = list(str(word_list[0]))
+def move_letters(word, index):
+    letters_to_move = word[:index]
 
-# handle the first word since it is capitalized
-if letter_list[0] in FIRST_LETTER_VOWELS:
-    if letter_list[-1] in PUNCTUATION:
-        pig_latin_word = ''.join(letter_list[:-1]) + 'ay' + letter_list[-1]
+    if word[-1] in PUNCTUATION:
```

Comment:
I think this could be further broken up into separate steps. You are effectively:

1. Detecting if there is ending punctuation
1. Doing the Pig Latin transformation
1. Fixing up any ending punctuation

But the middle step is repeated depending on whether there is punctuation. This could be streamlined with multiple functions and perhaps by marking the empty string as the punctuation if there is none. Using the empty string is a great trick.